### DISCUSSION TITLE
Confusion with execution date and timedelta.

### DISCUSSION THREAD

**Discussion by Vishal2696** [apache/airflow#30506] [Q&A]
I have a normal parent & child dag dependancy.The child dag monitors for a task to be completed successfully in the parent dag using `ExternalTaskSensor`. Below is the code for my Child dag's task. 
```
start_date = pendulum.datetime(2021, 6, 10, tz="Australia/Melbourne")
dag = DAG ("Child_dag_1", schedule_interval="0 15 * * *", default_args={'owner':'myapp', 'start_date': start_date}, catchup=False, dagrun_timeout=datetime.timedelta(minutes=1440))
parenttask_monitor_sensor = ExternalTaskSensor(
        task_id = "child_task_1",
        external_dag_id = "Parent_dag_1",
        external_task_id = "Parent_task_1",
        allowed_states = ["success"],
        execution_delta = datetime.timedelta(days=1),
        dag=dag
    )
```

This dependancy was running fine until April 2, 2023. On April 2, The parent dag's task finished successfully but the child dag's sensor could not identify this and just kept poking until we manually marked the task to success. I have troubleshooted and have found that there might be possible mismatch with the exec_date of the two dags but I want to confirm that here. Below is my findings. 

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/LOCAL_~1/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/LOCAL_~1/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

Date | Parent dag | Child dag
-- | -- | --
31/03/2023 | Exec_date: 31/03      4:00:00 | Exec_date: 31/03      4:00:00
1/04/2023 | Exec_date: 01/04      4:00:00 | Exec_date: 01/04      4:00:00
2/04/2023 | Exec_date: 02/04       5:00:00 | Exec_date: 02/04       5:00:00

</body>

</html>

As you can see from the above data, In the exec_date timefield there has been a 1 hour delay from April 2 onwards. This is because the daylight savings in AEST (Australia) ends on the 2nd of April 3 am. My theory is this 1 hour change messed up the timedelta as the parent dag's previous execution was less than 24 hours now (01/04 4:00:00). 
I want to know if my theory is correct or have I misunderstood the concept of exec date? 

If my theory is correct, then is there any way or recommendation to avoid such cases? 

_**Note:** From April 4th, onwards the dags were back to normal without any issues. (3rd April we paused the dags manually)_

**Comment by potiuk** [comment]
Looks like timezone issue, indeed. You have not written about your version, but yeah, I think there are still a few issues related to time-zone change that I am afraide are difficult to debug, and I think the most you can do is create a similar issue and describe in details what happens, possibly also try to extract logs from around the time it happened and clearly describe in the issue title that it is time-zone relateda. Likey we will have some more concerted effort before the next tz change to diagnose and fix it.
